### PR TITLE
Eval fixes for cross-compilation

### PR DESCRIPTION
**Pull request by Mindavi**
###### Description of changes

Fix evaluation of `nix-env -f . -qaP --attr pkgsCross.aarch64-multiplatform --out-path --show-trace` for a bunch of packages. It still doesn't fully evaluate, unfortunately.

Improves progress for https://github.com/NixOS/nixpkgs/issues/116207.

###### Things done

- Tried evaluating nixpkgs with the above function

Opening as draft for now since I'm still working at it, but I like sharing it already :)

### CODE REVIEW

**Review by jshcmpbll — COMMENTED** [review]
```zsh
mbazg40bpcmixjin0x2jxynnd4kram9q-azure-functions-core-tools-3.0.3785/bin ./func
Process terminated. Couldn't find a valid ICU package installed on the system. Set the configuration flag System.Globalization.Invariant to true if you want to run with no globalization support.
   at System.Environment.FailFast(System.String)
   at System.Globalization.GlobalizationMode.GetGlobalizationInvariantMode()
   at System.Globalization.GlobalizationMode..cctor()
   at System.Globalization.CultureData.CreateCultureWithInvariantData()
   at System.Globalization.CultureData.get_Invariant()
   at System.Globalization.CultureInfo..cctor()
   at System.StringComparer..cctor()
   at System.StringComparer.get_OrdinalIgnoreCase()
   at Azure.Functions.Cli.Program.SetCoreToolsEnvironmentVariables(System.String[])
   at Azure.Functions.Cli.Program.Main(System.String[])
[1]    95172 abort (core dumped)  ./func
```
Looks like ive got some other things to fix in this derivation 😞

**Review by corngood — APPROVED** [review]

**Review by SuperSandro2000 — COMMENTED** [review]
LGTM except I can't comment on the buildPackages part.

**Comment by Mindavi** [comment]
Let's leave it at this for now, it appears there's quite some things to go through, so let's not make this enormous. (Besides, it takes a lot of time and RAM to do an eval already).

**Comment by Mindavi** [comment]
Will need to drop a commit (the one for build-dotnet-module) due to this being merged: https://github.com/NixOS/nixpkgs/pull/173889

File: pkgs/tools/package-management/conda/default.nix [hunk 1]
```diff
@@ -42,7 +42,7 @@ let
         zlib # libz.so.1
       ];
     in
-      runCommand "conda-install" { buildInputs = [ makeWrapper zlib]; }
+      runCommand "conda-install" { buildInputs = [ buildPackages.makeWrapper zlib]; }
```
Comment (Artturin) [inline pkgs/tools/package-management/conda/default.nix:45]: `nativeBuildInputs` works in `runCommand`

**Comment by Artturin** [comment]
steam eval fix in https://github.com/NixOS/nixpkgs/pull/186512

**Comment by Mindavi** [comment]
Thanks for rebasing :)

**Review by corngood — APPROVED** [review]

**Comment by Artturin** [comment]
more https://github.com/NixOS/nixpkgs/pull/186564

**Pull request merged**